apps/prairielearn/src/middlewares/csrfToken.ts

```diff
@@ -1,13 +1,17 @@
-// @ts-check
 import asyncHandler from 'express-async-handler';
 
 import { HttpStatusError } from '@prairielearn/error';
 import { generateSignedToken, checkSignedToken } from '@prairielearn/signed-token';
 
 import { config } from '../lib/config.js';
 
+interface TokenData {
+  url: string;
+  authn_user_id?: string;
+}
+
 export default asyncHandler(async (req, res, next) => {
-  const tokenData = {
+  const tokenData: TokenData = {
     url: req.originalUrl,
   };
```

Comment:
Totally bikeshedding, but I think we can just rewrite this as the following, which is a little simpler and means we don't need a separate type:

```suggestion
  const tokenData: TokenData = {
    url: req.originalUrl,
    authn_user_id: res.locals.authn_user?.user_id,
  };
```